hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientWithKeyLatestVersion.java

```diff
@@ -0,0 +1,198 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.client.rpc;
+
+import org.apache.hadoop.hdds.client.ReplicationConfig;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.ScmConfigKeys;
+import org.apache.hadoop.ozone.MiniOzoneCluster;
+import org.apache.hadoop.ozone.client.BucketArgs;
+import org.apache.hadoop.ozone.client.ObjectStore;
+import org.apache.hadoop.ozone.client.OzoneBucket;
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.client.OzoneClientFactory;
+import org.apache.hadoop.ozone.client.OzoneVolume;
+import org.apache.hadoop.ozone.client.io.OzoneInputStream;
+import org.apache.hadoop.ozone.client.io.OzoneOutputStream;
+import org.apache.hadoop.ozone.om.helpers.OzoneFileStatus;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationType.RATIS;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_CLIENT_KEY_LATEST_VERSION_LOCATION;
+import static org.junit.Assert.fail;
+
+/**
+ * Main purpose of this test is with OZONE_CLIENT_KEY_LATEST_VERSION_LOCATION
+ * set/unset key create/read works properly or not for buckets
+ * with/with out versioning.
+ */
+@RunWith(Parameterized.class)
+public class TestOzoneRpcClientWithKeyLatestVersion {
+
+  private MiniOzoneCluster cluster;
+
+  private ObjectStore objectStore;
+
+  private OzoneClient ozClient;
+
+  private boolean getLatestVersion;
+
+
+  @Parameterized.Parameters
+  public static Collection<Object> data() {
+    return Arrays.asList(true, false);
+  }
+
+  public TestOzoneRpcClientWithKeyLatestVersion(boolean getLatestVersion) {
+    try {
+      setup(getLatestVersion);
+    } catch (Exception ex) {
+      fail("Unexpected exception during setup:" + ex);
+    }
+  }
+
+
+  public void setup(boolean latestVersion) throws Exception {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.setInt(ScmConfigKeys.OZONE_SCM_PIPELINE_OWNER_CONTAINER_COUNT, 1);
+    conf.setBoolean(OZONE_CLIENT_KEY_LATEST_VERSION_LOCATION, latestVersion);
```

Comment:
As far as I understand, this config (`ozone.client.key.latest.version.location`) only affects the client.  Can we avoid spinning up a new cluster each time?  This would halve test runtime.

(Cluster setup/shutdown can be done in `@BeforeClass`/`@AfterClass` methods.)

Reply:
Done